### DISCUSSION TITLE
Added Terrain editor to the engine

### DISCUSSION THREAD

**Discussion by VecterraSoft** [Hoimar/Planet-Generator#42] [General]
https://github.com/user-attachments/assets/c6dbe300-1e18-43ce-9601-da4ffda56d5a

Still need to work on a few details for edges.
Also added a shader that blends a different material for the cliffs as well as a macro and a detail texture, which are off on here.

**Comment by Hoimar** [comment]
Damn that is looking pretty cool! How did you do it?

**Reply by SonomSoft** [reply]
Thats an old version, i have since fixed the issues that i had before. I finished adding  terrain  sculpting and painting as well. Also added a new water shader.
Im going to read some tutorials on how to use git hub and post a fork of your engine.

![planetengine46](https://github.com/user-attachments/assets/c0a4e893-dd93-456c-ac7f-df1213f4d1b4)

**Reply by SonomSoft** [reply]
Also, do you by any chance know how i can fix the atmosphere fog that is broken in Godot 4.3 ? The atmosphere is s till there but the terrain no longer represents the fog.

**Reply by SonomSoft** [reply]
Forgot to answer the other question. The way i did the terrain sculpting was to create an array that represents the positions of the verts and then just modify the array and update the chunk. And in that same way you can edit the vert colors, just with a color array.

**Comment by SonomSoft** [comment]
A little more polished now. UI and stuff.  Added grass generation and tree propagation. NPC characters that do tasks. Medieval theme. Can terraform and create towns and castles.

![Genesis ss2](https://github.com/user-attachments/assets/baa8b290-1957-499f-af82-9ffffb80ffce)

![genesis ss3](https://github.com/user-attachments/assets/8afc466e-68ea-48c6-b945-104ae588322f)

![genesisss4](https://github.com/user-attachments/assets/c2e656de-e166-4c6c-9d6e-0fdcb64c3fa2)


![Screenshot from 2024-11-13 19-20-46](https://github.com/user-attachments/assets/b0584222-b618-49b4-a6ee-e1dfff8ac12f)

**Comment by SonomSoft** [comment]
Quick update to the new ocean shader.

![genesisEngine](https://github.com/user-attachments/assets/cd886a89-7927-4f5b-9467-2ea64b2228b1)

![ocean](https://github.com/user-attachments/assets/7ef1ecb0-9bd9-4b32-bf3b-531937f318b3)

**Comment by SonomSoft** [comment]
![genesisEngine2](https://github.com/user-attachments/assets/952bd3b0-f232-4bd8-b46b-af4e7b9bef16)
